workloads/upgrade-perf/rosa_upgrade.sh

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env bash
+export rosa_environment=${ROSA_ENVIRONMENT:=staging}
+# Nightly versions cannot be upgraded, only valid channels for upgrading are stable and candidate
+export rosa_version_channel=${ROSA_VERSION_CHANNEL:=stable}
+_es_index=${ES_INDEX:=managedservices-install-timings}
+
+rosa_upgrade(){
+  if [ ${rosa_version_channel} == "nightly" ] ; then
+    echo "ERROR: Invalid channel group. Nightly versions cannot be upgraded. Exiting..."
+    exit 1
+  fi
+  echo "To-Version: ${TOVERSION}"
+  echo "To-Latest: ${LATEST}"
+
+  echo "ROSA Cluster: ${ROSA_CLUSTER_NAME}"
+  echo "ROSA Environment: ${rosa_environment}"
+  echo "ROSA Channel Group: ${rosa_version_channel}"
+  rosa login --env=${rosa_environment} --token=${ROSA_TOKEN}
+  if [ $(rosa list clusters | awk '{print $2}' | sed -e 1d | grep ${ROSA_CLUSTER_NAME}) != ${ROSA_CLUSTER_NAME} ] ; then
+    echo "ERROR: Cluster ${ROSA_CLUSTER_NAME} not found on rosa list clusters results. Exiting..."
+    exit 1
+  fi
+  if [ ! -z ${LATEST} ] ; then
+    VERSION=$(rosa list upgrade -c ${ROSA_CLUSTER_NAME} | sed -e 1d | head -1 | awk '{print $1}')
+  elif [ ! -z ${TOVERSION} ] ; then
+    VERSION=$(rosa list upgrade -c ${ROSA_CLUSTER_NAME} | sed -e 1d | awk '{print $1}' | grep ${TOVERSION})
+  else
+    echo "ERROR: Missing upgrade strategy. No LATEST or TOVERSION are defined. Exiting..."
+    exit 1
+  fi
+  if [ -z ${VERSION} ] ; then
+    echo "ERROR: No version to upgrade found running rosa list upgrade -c ${ROSA_CLUSTER_NAME}"
+    exit 1
+  fi
+  CURRENT_VERSION=$(oc get clusterversion | grep ^version | awk '{print $2}')
+  rosa upgrade cluster -c ${ROSA_CLUSTER_NAME} --version=${VERSION} -y --schedule-date $(date +%Y-%m-%d) --schedule-time $(date -u --date="+6 minutes" +%H:%M)
+  # Sleep 6 minutes, rosa upgrade dont let to schedule an upgrade in less than 5 minutes from now
+  sleep 360
+  oc delete pods -n openshift-insights --all
+  oc delete pods -n openshift-managed-upgrade-operator --all
+  rosa_control_plane_upgrade_active_waiting ${VERSION}
+  if [ $? -eq 0 ] ; then
+    CONTROLPLANE_UPGRADE_RESULT="OK"
+  else
+    CONTROLPLANE_UPGRADE_RESULT="Failed"
```

Comment:
I don't think the upgrade proceeds after the control plane upgrade fails, we should exit here since it doesnt make sense to wait for worker to update, since the worker MCPs only update after the control plane updates. We just need to construct the ES docuemnt here and exit 1 from the script.

Reply:
Actually, we should wait for the workers, because on that phase is where we check the result of the Rosa upgrade. On that phase, we know that we are waiting for the workers to be restarted, but what we are checking is the result of the Rosa show process, which is the scope of the script, to check the time that Rosa upgrade command takes to finish

Reply:
OK. As per my previous experience doing OCP upgrades, the workers are only upgraded after the control plane succeeds, maybe its different with ROSA?